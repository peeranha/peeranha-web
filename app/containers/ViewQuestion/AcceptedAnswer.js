import React from 'react';
import PropTypes from 'prop-types';

import Content from './Content';
import { ANSWER_TYPE } from './constants';

export const AcceptedAnswer = /* istanbul ignore next */ props => (
  <React.Fragment>
    {props.questionData.answers.map(
      item =>
        props.questionData.correct_answer_id === item.id ? (
          <Content
            {...props}
            className="mb-3"
            key={`${ANSWER_TYPE}${item.id}`}
            answerId={item.id}
            comments={item.comments}
            content={item.content}
            rating={item.rating}
            isTheLargestRating={item.isTheLargestRating}
            questionFrom={props.questionData.user}
            isItWrittenByMe={item.isItWrittenByMe}
            history={item.history}
            userInfo={item.userInfo}
            postTime={item.post_time}
            lastEditedDate={item.lastEditedDate}
            votingStatus={item.votingStatus}
            deleteItem={props.deleteAnswer}
            editItem={props.editAnswer}
            saveComment={props.saveComment}
            deleteComment={props.deleteComment}
            buttonParams={{
              questionId: props.questionData.id,
              answerId: item.id,
              whowasvoted: item.userInfo.user,
            }}
          />
        ) : null,
    )}
  </React.Fragment>
);

AcceptedAnswer.propTypes = {
  questionData: PropTypes.object,
  deleteAnswer: PropTypes.func,
  editAnswer: PropTypes.func,
  saveComment: PropTypes.func,
  deleteComment: PropTypes.func,
};

export default React.memo(AcceptedAnswer);
