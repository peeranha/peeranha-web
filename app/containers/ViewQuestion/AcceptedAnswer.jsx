import React from 'react';
import PropTypes from 'prop-types';
import * as routes from 'routes-config';

import Content from './Content';
import { ANSWER_TYPE } from './constants';

export const AcceptedAnswer = props => (
  <React.Fragment>
    {props.questionData.answers.map(
      item =>
        props.questionData.correct_answer_id === item.id ? (
          <Content
            {...props}
            className="mb-3"
            key={`${ANSWER_TYPE}${item.id}`}
            answerId={item.id}
            comments={item.comments}
            content={item.content}
            rating={item.rating}
            isTheLargestRating={item.isTheLargestRating}
            questionFrom={props.questionData.user}
            isItWrittenByMe={item.isItWrittenByMe}
            history={item.history}
            userInfo={item.userInfo}
            postTime={item.post_time}
            lastEditedDate={item.lastEditedDate}
            votingStatus={item.votingStatus}
            deleteItem={props.deleteAnswer}
            deleteItemLoading={props.deleteAnswerLoading}
            editItem={[
              props.redirectToEditAnswerPage,
              routes.answerEdit(props.questionData.id, item.id),
            ]}
            saveComment={props.saveComment}
            deleteComment={props.deleteComment}
            buttonParams={{
              questionId: props.questionData.id,
              answerId: item.id,
              whowasvoted: item.userInfo.user,
            }}
          />
        ) : null,
    )}
  </React.Fragment>
);

AcceptedAnswer.propTypes = {
  questionData: PropTypes.object,
  deleteAnswer: PropTypes.func,
  saveComment: PropTypes.func,
  deleteComment: PropTypes.func,
  redirectToEditAnswerPage: PropTypes.func,
  deleteAnswerLoading: PropTypes.bool,
};

export default React.memo(AcceptedAnswer);
