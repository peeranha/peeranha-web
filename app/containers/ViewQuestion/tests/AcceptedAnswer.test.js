import AcceptedAnswer from '../AcceptedAnswer';

let props;

beforeEach(() => {
  props = {
    questionData: {
      answers: [
        {
          id: 1,
        },
        {
          id: 2,
        },
        {
          id: 3,
        },
      ],
      correct_answer_id: 1,
    },
  };
});

describe('AcceptedAnswer', () => {
  it('test, correct_answer_id !== item.id', () => {
    props.questionData.correct_answer_id = 0;
    expect(AcceptedAnswer(props)).toMatchSnapshot();
  });

  it('test, correct_answer_id === item.id', () => {
    props.questionData.correct_answer_id = 1;
    expect(AcceptedAnswer(props)).toMatchSnapshot();
  });
});
