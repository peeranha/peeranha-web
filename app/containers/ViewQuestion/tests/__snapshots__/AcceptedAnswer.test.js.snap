// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptedAnswer test, correct_answer_id !== item.id 1`] = `<React.Fragment />`;

exports[`AcceptedAnswer test, correct_answer_id === item.id 1`] = `
<React.Fragment>
  <UNDEFINED
    answerId={1}
    buttonParams={
      Object {
        "answerId": 1,
        "questionId": undefined,
        "whowasvoted": undefined,
      }
    }
    className="mb-3"
    questionData={
      Object {
        "answers": Array [
          Object {
            "id": 1,
            "userInfo": Object {},
          },
          Object {
            "id": 2,
            "userInfo": Object {},
          },
          Object {
            "id": 3,
            "userInfo": Object {},
          },
        ],
        "correct_answer_id": 1,
      }
    }
    userInfo={Object {}}
  />
</React.Fragment>
`;
